### PR TITLE
Guide new users to add use `super::*;` to `mod test`

### LINKED ISSUE #10559: Guide new users to add `use super::*;` to `mod test`.
### Problem

`cargo init --lib` produces a src/lib.rs file contaning this:
```rust
#[cfg(test)]
mod tests {
    #[test]
    fn it_works() {
        let result = 2 + 2;
        assert_eq!(result, 4);
    }
}
```
When I was first learning Rust, I'd start with this code, and add a new function, and a unit test, like this:
```rust
pub fn add_numbers(a: i32, b: i32) -> i32 {
    a + b
}

#[cfg(test)]
mod tests {
    #[test] 
    fn it_works() {
        let result = add_numbers(2, 2);
        assert_eq!(result, 4);
    }
}
```
But, this doesn't compile:
```
error[E0425]: cannot find function `add_numbers` in this scope
 --> src/lib.rs:9:22
  |
9 |         let result = add_numbers(2, 2);
  |                      ^^^^^^^^^^^ not found in this scope
  |
help: consider importing this function
  |
8 |     use crate::add_numbers;
  |
```
So first, this in itself is a mild papercut&mdash;the seemingly obvious thing didn't work. But next, if I would follow rustc's suggestions, or let IDEs fix this for me, I'd eventually end up with a big pile of `use`s, as I write more code and add tests for it. This would add more papercuts: clutter, and a focal point for merge conflicts.

It would almost always be better for me to use `use super::*` inside the `mod tests`, because once that's added, it avoids these papercuts from then on. So the feature request is: can we do more to guide users to `use super::*;`?

The [relevant documentation](https://doc.rust-lang.org/book/ch11-01-writing-tests.html) does recommend using `use super::*;`, but when I was learning Rust, I didn't read that section because I knew what an "assert" was from other languages and didn't think I needed to read that whole section in the docs, so I missed the `use super::*;` advice within.

As a historical note, these papercuts contributed to my forming a habit of writing unit tests without `mod tests`, which is what prompted the discussion which led to #10531.

### Proposed Solution

Adding `use super::*;` to the current code would get an unused-import warning. So one option would be to add it, but commented out, like this:

```rust
#[cfg(test)]
mod tests {
    // Uncomment this to use items from the containing module.
    //use super::*;

    #[test]
    fn it_works() {
        let result = 2 + 2;
        assert_eq!(result, 4);
    }
}
```

### Notes

_No response_

### PR DESCRIPTION
**Pull request by merelymyself**
### What does this PR try to resolve?

Currently, `cargo init --lib` produces examples for unit tests. However, new users will find that they are unable to use functions they defined above. This should resolve the confusion.

Fixes #10559 

### How should we test and review this PR?

This PR does not create new features. Test this PR using the already-existing tests.

### Additional information

I didn't think this was a major change, so I did not open a RFC for it. Please let me know if I should have!

### CODE REVIEW

File: src/cargo/ops/cargo_new.rs [hunk 1]
```diff
@@ -811,6 +811,8 @@ fn main() {
             b"\
 #[cfg(test)]
 mod tests {
+    // Uncomment this to use items from the containing module.
+    // use super::*;
```
Comment (Muscraft) [inline src/cargo/ops/cargo_new.rs:815]: Small nitpick, this might be better as `//use super::*;`. The extra space could get missed

**Comment by Muscraft** [comment]
Thinking about this some more, I think adding something for users to uncomment only solves a symptom of the issue and not the issue itself. I think the main issue is the error message from rustc ([which appears to be generated in this function](https://github.com/rust-lang/rust/blob/4f68efad64f6a54703521d465817b6103813694d/compiler/rustc_resolve/src/diagnostics.rs#L2444-L2447)). It might be a better idea (and a harder change), to update the error message to include `use super::*;`.

Something like:
```
error[E0425]: cannot find function `add_numbers` in this scope
 --> src/lib.rs:9:22
   |
9  |         let result = add_numbers(2, 2);
   |                      ^^^^^^^^^^^ not found in this scope
   |
help: consider importing one of these items
   |
LL | use super::*;
   |
LL | use crate::add_numbers;
   |
```
I mocked this from [here](https://github.com/rust-lang/rust/blob/4f68efad64f6a54703521d465817b6103813694d/src/test/ui/did_you_mean/issue-56028-there-is-an-enum-variant.stderr#L19-L35)

I think one of the maintainers may have more insight than I do on the route to go about this

File: src/cargo/ops/cargo_new.rs (hunk 1 above)
Comment (epage) [inline src/cargo/ops/cargo_new.rs:815]: I'm a bit mixed about how we solve the original concern.  I've posted my thoughts to https://github.com/rust-lang/cargo/issues/10559

**Review by epage — CHANGES_REQUESTED** [review]

**Comment by merelymyself** [comment]
@epage - followed your second suggestion, thought it was the clearest. Thank you!

File: src/cargo/ops/cargo_new.rs [hunk 2]
```diff
@@ -809,11 +809,17 @@ fn main() {
...
         } else {
             b"\
+pub fn add(left: usize, right: usize) -> usize {
+    left + right
+}
+
 #[cfg(test)]
 mod tests {
+    use super::*;
+
```
Comment (Muscraft) [inline src/cargo/ops/cargo_new.rs:819]: It looks like there is extra whitespace here that is causing the error. That is a super odd thing.
In `src/cargo/ops/cargo_new.rs`
Screen Shot 2022-05-27 at 11 06 38
in `tests/testsuite/init/auto_git/out/src/lib.rs`
Screen Shot 2022-05-27 at 11 06 54
From failed tests
Screen Shot 2022-05-27 at 11 09 10

**Review by Muscraft — CHANGES_REQUESTED** [review]

File: src/cargo/ops/cargo_new.rs (hunk 2 above)
Comment (merelymyself) [reply src/cargo/ops/cargo_new.rs:819]: Yep, thanks for pointing it out!

**Review by epage — APPROVED** [review]
I think I'm fine with this.  The new template does not make a material difference in the boiletplate a user has to delete while it better teaches a user how the existing boilerplate would work.

I'm assuming this is small enough not to need an FCP but if anyone from the cargo team disagrees, we can always revert it and go through an FCP

**Pull request merged**